### ISSUE TITLE
negative clock starting on low shows phantom edge

### ISSUE THREAD

**Issue by jeremyherbert** [wavedrom/wavedrom#304]
Is it possible to have a negative clock with "N", but that starts on a low signal? For example, wavedrom shows this with "0...N......"

![WaveDrom_Editor](https://user-images.githubusercontent.com/52499/91299427-eba15500-e7e4-11ea-8f7a-ba1fecfee52c.png)

But what I am trying to show is this:

<img width="120" alt="Saleae_Logic_Software" src="https://user-images.githubusercontent.com/52499/91299469-f825ad80-e7e4-11ea-87b5-11ef7d08b59a.png">

**Comment by drom** [comment]
Agree, that is the bug.

**Comment by Remillard** [comment]
Yep, I just ran into this as well.  SPI discontinuous clock with polarity 0 starts low and then has a rising edge mid bit period with no glitch.  I think it's going to be back to Visio for the time being, but hopefully this can be resolved.

**Comment by drom** [comment]
How about this solution?

```
{signal: [
  {name: 'clk1', wave: 'l..N..'},
  {name: 'clk2', wave: 'l.hN..'},
  {name: 'clk3', wave: 'h..P..'},
  {name: 'clk4', wave: 'h.lP..'},
]}
```
![image](https://user-images.githubusercontent.com/511872/118586459-a399d800-b74f-11eb-9c1e-2c1115a21f04.png)

`clk1` : Before the `Negative edge` you need to un-gate clock into `high` state.
`clk2` : that can be done with symbol `h` in the cycle right before the edge.

`clk3` : Before the `Positive edge` you need to un-gate clock into `low` state.
`clk4` : that can be done with symbol `l` in the cycle right before the edge.

**Comment by Remillard** [comment]
I don't think that would work for me.  I needed this output:
![image](https://user-images.githubusercontent.com/28012313/118665517-fa3afc80-b7b7-11eb-8e7d-7b64dba1e3ef.png)

`sclk` is low and then its first transition is low to high as the capturing edge in the middle of the bit period.  In none of the examples above would this be conveyed appropriately.  Hopefully at some point wavedrom will transition from 0 into a low to high clock without the glitch edge.

**Comment by drom** [comment]
@Remillard will this work?

```
{signal: [
  {name: 'sclk',   wave: 'l.P.|.', phase: .5},
  {name: 'sdmosi', wave: '022<2|>2', data: 'WR A14 A13 A12'},
]}
```
![image](https://user-images.githubusercontent.com/511872/118673229-afb48280-b7ad-11eb-89d5-502e318cec70.png)

**Comment by Remillard** [comment]
That is more appropriate, though it still leaves half a phase at the end.  I didn't show the entire waveform but there's the usual SPI stuff with chip select deasserting and then another transaction afterwards.  The extra phase might make it difficult to align these things and in any case would need post processing to crop the trailing phase.

There's no need for an immediate solution.  I drew the picture in Visio and got on with my life.  I'm just adding support for the tool to properly handle a discontinuous clock without glitches.  Presumably the same problem would happen if you had a positive going clock and the clock was low afterwards.  I would bet there's another line going upwards at the end of the positive clock glyph.

I would suggest making another glyph of each clock without the leading/trailing vertical line and any time you have a solid logic level transitioning into the clock, you use the special glyph (where applicable -- obviously a logic high going into a negative phase clock would want the original glyph while a logic low going into a negative phase clock would want the new glyph) and the clock transitioning into a level state (with the same caveat).

It's a good tool, there are just many corner cases to cover when dealing with timing diagrams.  Wavedrom does a pretty good job with most of them.  Discontinuous clocks are just still a problem area.

**Comment by Remillard** [comment]
Also is there better documentation than https://wavedrom.com/tutorial.html?  I did not recognize that `<` code.  Several years ago I was looking for something like that where it would control how wide a data phase followed by a discontinuity code.  Didn't know about it.

**Comment by suzizecat** [comment]
Hi there, I got exactly on the same situation and the best I could get was this : 
```json
{
  "signal": [
    {"name": "\\symb{csn}",  "wave": "10.."                   },
    {"name": "\\symb{clk}",  "wave": ".<lh>NN"                },
    {"name": "\\symb{mosi}", "wave": "x222",  "data": "1 2 3" },
    {"name": "\\symb{miso}", "wave": "x222",  "data": "1 2 3" }
  ],
  "foot": {
    "tick": 0
  }
}
```
![image](https://user-images.githubusercontent.com/1909563/179712946-5b95c294-52b1-43e1-b125-bba1d5c8a442.png)

My take on this is that using `"0<.h>` should do the trick, but the dot in first position will not capture the `0`from outside the group and will default to `X`. Likewise, using the "group" (`<>` syntax) at the beggining of a wave will be treated as X. (otherwise `<0..h>L` would have done the trick.

I opened #368 related to this point.
